app/jobs/opennebula/compute_action_job.rb

```diff
@@ -5,35 +5,91 @@ class ComputeActionJob < OpennebulaActionJob
     INACTIVE_ACTIONS = %w[start save].freeze
     ACTIONS = [ACTIVE_ACTIONS, INACTIVE_ACTIONS].flatten.freeze
 
+    # Template/image attributes to remove after save
+    PURGE_ATTRIBUTES = %w[TEMPLATE/CLOUDKEEPER_ID TEMPLATE/CLOUDKEEPER_EXPIRED].freeze
+
     # :nodoc:
     ELM_NAME = 'virtual_machine'.freeze
 
+    protected
+
+    # :nodoc:
+    def job_cleanup!
+      super
+      return unless @_cleanup_templates
+
+      logger.debug { "Running after-fail cleanup on #{@_cleanup_templates.inspect}" }
+      @_cleanup_templates.each do |tpl|
+        logger.warn "Cleaning up TEMPLATE##{tpl} after save action failed"
+        begin
+          handle { template(tpl).delete(true) }
+        rescue StandardError => ex
```

Comment:
Are you sure you should be catching `StandardError`?

Reply:
Yes.